### ISSUE TITLE
How did you prepare the manifest dir for pretrain and in which format?

### ISSUE THREAD

**Issue by sanjuktasr** [k2-fsa/icefall#1705]
Hi there @yfyeung , first of all thank you for creating this SSL recipe!

I tried running your zipformer/ codes, but my model diverged at epoch 33 and pretraining ended with a `Grad scale is small` error. 

Throughout pretraining before the divergence, I noticed my `grad scale` tended to fluctuate between 0.125 and 2. 

Did you face the same issues? 

EDIT: I was also wondering if you tried toggling the loss reduction to `mean` instead of `sum`. Maybe that will stabilise training? 

My commands. I adapted the batch size to my setup, maintaining the same `accum_grad * max_duration * world_size`. 
```bash
# pretraining
python zipformer/pretrain.py \
    --world-size 4 \
    --use-fp16 1 \
    --num-epochs 50 \
    --manifest-dir data/raw \
    --max-duration 350 \
    --accum-grad 2 \
    --exp-dir zipformer/exp2/pretrain
```

As per your explanation, I used the same 500 k-means labels from `simple_kmeans`.

_Originally posted by @teowenshen in https://github.com/k2-fsa/icefall/issues/1500#issuecomment-2050792777_

**Comment by yfyeung** [comment]
> ```
>           Hi there @yfyeung , first of all thank you for creating this SSL recipe!
> ```
> 
> I tried running your zipformer/ codes, but my model diverged at epoch 33 and pretraining ended with a `Grad scale is small` error.
> 
> Throughout pretraining before the divergence, I noticed my `grad scale` tended to fluctuate between 0.125 and 2.
> 
> Did you face the same issues?
> 
> EDIT: I was also wondering if you tried toggling the loss reduction to `mean` instead of `sum`. Maybe that will stabilise training?
> 
> My commands. I adapted the batch size to my setup, maintaining the same `accum_grad * max_duration * world_size`.
> 
> ```shell
> # pretraining
> python zipformer/pretrain.py \
>     --world-size 4 \
>     --use-fp16 1 \
>     --num-epochs 50 \
>     --manifest-dir data/raw \
>     --max-duration 350 \
>     --accum-grad 2 \
>     --exp-dir zipformer/exp2/pretrain
> ```
> 
> As per your explanation, I used the same 500 k-means labels from `simple_kmeans`.
> 
> _Originally posted by @teowenshen in [#1500 (comment)](https://github.com/k2-fsa/icefall/pull/1500#issuecomment-2050792777)_

Hi, mean is used to simulate multi-GPU training, and sum is used to simulate a large batch size. I used 8 GPUs with a max duration of 600. It seems that maintaining the same accum_grad * max_duration * world_size doesn't match the original setup.
